sklearn/utils/estimator_checks.py

```diff
@@ -290,6 +298,9 @@ def check_pipeline_consistency(name, Estimator):
                       random_state=0, n_features=2, cluster_std=0.1)
     X -= X.min()
     y = multioutput_estimator_convert_y_2d(name, y)
+    if name == "SpectralEmbedding":
+        # This data makes sure that the graph is fully connected.
+        X, y = S_X, S_y
```

Comment:
Currently 2 things cause the o/p to be non-deterministic...

Major one is when the data produces an adjacency graph which is not fully connected. (But a warning is raised stating spectral embedding may not behave correctly since the adj. graph is not fully connected, so I am not quite sure if non-deterministic output is to be expected here)

Minor one is when there are multiple max abs values where the sign has to be arbitrarily given to the first index etc... (I'll fix this and push the commit soon...)